### DISCUSSION TITLE
Where is the best place to store EAN numbers

### DISCUSSION THREAD

**Discussion by mittkkoo** [inventree/InvenTree#5682] [Q&A]
I'm currently exploring inventree. Many of our suppliers use EAN as product ID, along with their internal part number and the manufacturer's number. I recently found out that EAN is also quite commonly used in DATANORM datasets. 

Searching for a place for it in inventree I first thought it belongs to the manufacturer part. However I found that there I can only enter a custom parameter (without parameter template) and call it EAN. But this doesn't seem to be searchable and is even not visible in the UI. The next place would be the suppliers part. I can use parameter template there and the entry is searchable. But there seems not to be a table view where I can display a column with supplier's parts parameters, like the one for part parameters. I don't want to put the EAN in a part parameter, because I thought parts are not always identical with products (= manufacturer parts), i.e. there can be many equivalent products / supplier parts with different EAN associated with the same part.

Does anyone use the 13 digits EAN in inventree? How do you handle them?

**Comment by matmair** [comment]
We have EANs in parameters and use a barcode plugin for scanning integration.

**Comment by SchrodingersGat** [comment]
@mittkkoo in terms of improving "visibility" of EAN, I guess we have a few options:

- Add it as a first-class field to the `SupplierPart` model
- Add a table for viewing / searching / supplier part parameters
- Allow parameter values to be "searched"

Note that none of these are currently implemented :)

**Reply by wolflu05** [reply]
That was exactly, Why I opened #5576
